certificatesHandler_test.go

```diff
@@ -109,29 +106,78 @@ func TestGetUsersCertificates(t *testing.T) {
 	if res == nil {
 		t.Fatalf("Response is nil")
 	}
-	defer res.Body.Close()
 
 	//Certificates in memory for user A
-	a.InitInMemoryData()
 	var usersCertificates []Certificate
 	usersCertificates = append(usersCertificates, certificates[0], certificates[1])
 
 	// Certificates from response body
 	var respCertificates []Certificate
+
+	defer res.Body.Close()
 	if err := json.NewDecoder(res.Body).Decode(&respCertificates); err != nil {
 		t.Fatalf("Could not decode body: %v", err)
 	}
+
 	rcb, err := json.Marshal(respCertificates)
 	if err != nil {
 		t.Fatalf("Could not marshal into JSON: %v", err)
 	}
+
 	ucb, err := json.Marshal(usersCertificates)
 	if err != nil {
 		t.Fatalf("Could not marshal into JSON: %v", err)
 	}
+
 	assert.JSONEq(t, string(ucb), string(rcb))
 
 	//Clean up in memory data
 	certificates = certificates[:0]
 	users = users[:0]
 }
+
+func TestUpdateCertificate(t *testing.T) {
+	var jsonStrCertificate = []byte(`{        
+		"title": "Updated Title",
+		"note" : "Updated Note"
+	}`)
+	req := newRequest(t, "PATCH", "http://localhost:8000/certificates/1", bytes.NewBuffer(jsonStrCertificate))
+	req.SetBasicAuth("userA", "")
+	rec := executeRequest(req)
+
+	a.UpdateCertificate(rec, req)
+	checkResponseCode(t, http.StatusOK, rec.Code)
+
+	res := rec.Result()
+	if res == nil {
+		t.Fatalf("Response is nil")
+	}
+
+	//Update certificate 1
+	certificates[0].Title = "Updated Title"
+	certificates[0].Note = "Updated Note"
+
+	//Response certificate
+	var rUpdatedCertificate Certificate
+
+	defer res.Body.Close()
+	if err := json.NewDecoder(res.Body).Decode(&rUpdatedCertificate); err != nil {
+		t.Fatalf("Could not decode JSON: %v", err.Error())
+	}
+
+	rcb, err := json.Marshal(rUpdatedCertificate)
+	if err != nil {
+		t.Fatalf("Could not marshal into JSON: %v", err.Error())
+	}
+
+	ecb, err := json.Marshal(certificates[0])
+	if err != nil {
+		t.Fatalf("Could not marshal into JSON: %v", err.Error())
+	}
+
+	assert.JSONEq(t, string(ecb), string(rcb))
+
+	//Clean up in memory data
```

Comment:
I would put this reset into it's own function. Or better yet look at the go testing docs and see if there is a reset convention. Some function that will be called after every test.

Reply:
https://stackoverflow.com/questions/42310088/test-setup-and-teardown-for-each-test-in-go